apps/heft/src/plugins/TypescriptPlugin/TypescriptBuilder.ts

```diff
@@ -0,0 +1,1029 @@
+          // The TSLint config file doesn't exist - so just disable linting
+          this._lintingEnabled = false;
+        } else {
+          throw e;
+        }
+      }
+    }
+  }
+
+  public async invokeAsync(): Promise<void> {
+    const ts: ExtendedTypescript = require(this._configuration.typescriptToolPath);
+
+    const tslint: typeof Tslint | undefined = this._lintingEnabled
+      ? require(this._configuration.tslintToolPath)
+      : undefined;
+
+    this._terminal.writeLine(`Using TypeScript version ${ts.version}`);
+    if (tslint) {
+      this._terminal.writeLine(`Using TSLint version ${tslint.Linter.VERSION}`);
+    }
+
+    ts.performance.enable();
+
+    const measureTsPerformance: PerformanceMeasurer = <TResult extends object | void>(
+      measurementName: string,
+      fn: () => TResult
+    ) => {
+      const beforeName: string = `before${measurementName}`;
+      ts.performance.mark(beforeName);
+      const result: TResult = fn();
+      const afterName: string = `after${measurementName}`;
+      ts.performance.mark(afterName);
+      ts.performance.measure(measurementName, beforeName, afterName);
+      return {
+        ...result,
+        duration: ts.performance.getDuration(measurementName),
+        count: ts.performance.getCount(beforeName)
+      };
+    };
+
+    const measureTsPerformanceAsync: PerformanceMeasurerAsync = async <TResult extends object | void>(
+      measurementName: string,
+      fn: () => Promise<TResult>
+    ) => {
+      const beforeName: string = `before${measurementName}`;
+      ts.performance.mark(beforeName);
+      const resultPromise: Promise<TResult> = fn();
+      const result: TResult = await resultPromise;
+      const afterName: string = `after${measurementName}`;
+      ts.performance.mark(afterName);
+      ts.performance.measure(measurementName, beforeName, afterName);
+      return {
+        ...result,
+        duration: ts.performance.getDuration(measurementName)
+      };
+    };
+
+    if (this._configuration.watchMode) {
+      await this._runWatch(ts, measureTsPerformance);
+    } else {
+      await this._runBuild(ts, tslint, measureTsPerformance, measureTsPerformanceAsync);
+    }
+  }
+
+  public async _runWatch(ts: ExtendedTypescript, measureTsPerformance: PerformanceMeasurer): Promise<void> {
+    //#region CONFIGURE
+    const { duration: configureDurationMs, tsconfig, compilerHost } = measureTsPerformance(
+      'Configure',
+      () => {
+        const _tsconfig: Typescript.ParsedCommandLine = this._loadTsconfig(ts);
+        const _compilerHost: TWatchCompilerHost = this._buildWatchCompilerHost(ts, _tsconfig);
+        return {
+          tsconfig: _tsconfig,
+          compilerHost: _compilerHost
+        };
+      }
+    );
+    this._terminal.writeVerboseLine(`Configure: ${configureDurationMs}ms`);
+    //#endregion
+
+    this._validateTsconfig(ts, tsconfig);
+
+    ts.createWatchProgram(compilerHost);
```

Comment:
Is there a way to watch for changes in `tsconfig.json`?  Currently that's ignored.

Reply:
I think that'll be pretty difficult to handle. Maybe that's a feature we can add later.